common/src/main/java/io/netty/util/internal/ThrowableUtil.java

```diff
@@ -26,4 +30,26 @@ private ThrowableUtil() { }
         cause.setStackTrace(new StackTraceElement[] { new StackTraceElement(clazz.getName(), method, null, -1)});
         return cause;
     }
+
+    /**
+     * Gets the stack trace from a Throwable as a String.
+     *
+     * @param cause the {@link Throwable} to be examined
+     * @return the stack trace as generated by {@link Throwable#printStackTrace(java.io.PrintWriter)} method.
+     */
+    public static String stackTraceToString(Throwable cause) {
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        PrintStream pout = new PrintStream(out);
+        cause.printStackTrace(pout);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Use a logger to log this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1148)